tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -368,7 +366,8 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
             }
             else
             {
-                // parent is not a SpanContext, start a new trace
+                // if parent is not a SpanContext, it must be a custom ISpanContext implementation (its a public interface),
+                // assume we are starting a new trace since we don't have a TraceContext
```

Comment:
> it must be a custom ISpanContext implementation

Not necessarily right, we have that in version-conflict scenarios too right?

Reply:
In version conflict scenarios, we create a `SpanContext` implementation from the version-mismatch "distributed trace", so it would go down the branches:

```csharp
if (parent is SpanContext parentSpanContext)
{
    // ...
    if (traceContext == null)
    {
      // Logic here
    }
}
```

Reply:
What Zach said. The only other `ISpanContext` implementation in the tracer is `ReadOnlySpanContext` which, like any user-defined `ISpanContext`, doesn't have the properties we are interested in above.